Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.cpp

```diff
@@ -291,51 +306,53 @@ ActsExamples::ProcessCode ActsExamples::VertexPerformanceWriter::writeT(
         }
 
         const auto& majorityParticle = *it;
-        associatedTruthParticles.emplace_hint(associatedTruthParticles.end(),
-                                              majorityParticle);
+        trackParameters.push_back(trajectories.trackParameters(tip));
+        associatedTruthParticles.push_back(majorityParticle);
       }
     }
   }
 
-  ACTS_VERBOSE(
-      "Total number of reconstructed tracks : " << trackParameters.size());
-
   // Get number of track-associated true primary vertices
   m_nVtxReconstructable =
-      getNumberOfReconstructableVertices(associatedTruthParticles);
+      getNumberOfReconstructableVertices(SimParticleContainer(
```

Comment:
It would be handy to move getting the number of ReconstructableVertices in a vertexing helpers class to allow other algorithms to use it. I'm Ok to do that in a subsequent PR.

Reply:
I agree - that should be accessible outside the writer. same is true for the truth matching itself. we can do this after this pr